src/scanner.c

```diff
@@ -2122,29 +2122,28 @@ void GetChar ( void )
   GET_CHAR();
 
   /* handle escape equences                                              */
-  if ( *TLS(In) == '\\' ) {
-    GET_CHAR();
-    TLS(Value)[0] = GetEscapedChar();
-  }
-  else if ( *TLS(In) == '\n' ) {
-    SyntaxError("Newline not allowed in character literal");
-  }
-  /* put normal chars into 'TLS(Value)'                                  */
-  else {
-    TLS(Value)[0] = *TLS(In);
-  }
-
-  /* read the next character                                             */
-  GET_CHAR();
-
-  /* check for terminating single quote                                  */
-  if ( *TLS(In) != '\'' )
-    SyntaxError("Missing single quote in character constant");
+  if ( *TLS(In) == '\n' ) {
+    SyntaxError("Character literal must not include <newline>");
+  } else {
+    if ( *TLS(In) == '\\' ) {
+      GET_CHAR();
+      TLS(Value)[0] = GetEscapedChar();
+    } else {
+      /* put normal chars into 'TLS(Value)' */
+      TLS(Value)[0] = *TLS(In);
+    }
+    TLS(Symbol) = S_CHAR;
```

Comment:
The new code sets `Symbol` to `S_CHAR` *before* invoking `GET_CHAR`. And in particular, before checking for the "missing single quote character" error.  Does that have any significance? (I am not familiar with this code, not saying it is wrong, just wondering).